spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -199,6 +199,9 @@ public SparkTable alterTable(Identifier ident, TableChange... changes) throws No
           setSnapshotId = set;
         } else if ("cherry-pick-snapshot-id".equalsIgnoreCase(set.property())) {
           pickSnapshotId = set;
+        } else if ("sort-order".equalsIgnoreCase(set.property())) {
+          throw new UnsupportedOperationException("'sort-order' is a reserved table property. Please use the command " +
+                  "'ALTER TABLE ... WRITE ORDERED BY' to specify it.");
```

Comment:
One nit here I forgot about this before but we usually structure error messages as
"Cannot X because Y. Then the recommendation goes here"

I would also recommend not using "it" in the message since it the pronoun is a bit ambiguous. "to specify write sort-order" may be clearer

Reply:
Sorry I didn't note this before, but I think if we improve this message we can do the merge today

Reply:
NP. A new commit submitted for the comment. Thanks for the review.